libs/modelviewer/src/components/tags-overlay/TagsOverlay.tsx

```diff
@@ -25,58 +25,56 @@ export const TagsOverlay = (): JSX.Element => {
     setSelected(tag);
   };
 
-  // useEffect(() => {
-  //   console.log(filterTags);
-  // }, [filterTags]);
-
   return (
     <div>
-      {overlayTags.map((tag, index) => {
-        const isSelected = selected === tag.tagNo;
-        return (
-          <div
-            key={`${tag.tagNo}_${index}`}
-            title={titleResolver ? titleResolver(tag) : tag.tagNo}
-            onClick={() => {
-              echoInstance?.viewer.cameraManager.fitCameraToBoundingBox(
-                tag.boundingBox,
-                defaultRadiusFactor
-              );
-            }}
-          >
-            <RevealHtmlOverlayWrapper
-              overlayTool={overlayTool.current}
-              position3d={tag.position}
-              tagNo={tag.tagNo}
-              aabb={tag.aabb}
+      {overlayTags
+        .filter((ot) => filterTags.includes(ot.tagNo))
+        .map((tag, index) => {
+          const isSelected = selected === tag.tagNo;
+          return (
+            <div
+              key={`${tag.tagNo}_${index}`}
```

Comment:
Dont use index as keys in react. https://dev.to/shiv1998/why-not-to-use-index-as-key-in-react-lists-practical-example-3e66